### ISSUE TITLE
AutoGraph could not transform ... (Bad argument number for Name: 3, expecting 4)

### ISSUE THREAD

**Issue by bersbersbers** [tensorflow/tensorflow#35819]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows, Linux
- TensorFlow installed from (source or binary): pip
- TensorFlow version (use command below): 2.1.0
- Python version: 3.7.6

**Describe the current behavior**
Warning said to report to TensorFlow team, see below.

**Code to reproduce the issue**
```
import tensorflow as tf

def test(data):
    tf.cond(
        tf.less(tf.random.uniform((), maxval=2), 1),
        lambda: 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0,
        lambda: 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0,
    )
    return data

tf.data.Dataset.from_tensor_slices([[[0]]]).map(test)
```
**Other info / logs**
```
2020-01-13 16:01:26.037936: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cudart64_101.dll
2020-01-13 16:01:28.948415: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library nvcuda.dll
2020-01-13 16:01:29.117581: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1555] Found device 0 with properties:
pciBusID: 0000:02:00.0 name: Quadro M5000 computeCapability: 5.2
coreClock: 1.038GHz coreCount: 16 deviceMemorySize: 7.50GiB deviceMemoryBandwidth: 196.99GiB/s
2020-01-13 16:01:29.132706: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cudart64_101.dll
2020-01-13 16:01:29.150465: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cublas64_10.dll
2020-01-13 16:01:29.169042: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cufft64_10.dll
2020-01-13 16:01:29.177916: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library curand64_10.dll
2020-01-13 16:01:29.197289: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cusolver64_10.dll
2020-01-13 16:01:29.210804: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cusparse64_10.dll
2020-01-13 16:01:29.230095: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cudnn64_7.dll
2020-01-13 16:01:29.238015: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1697] Adding visible gpu devices: 0
2020-01-13 16:01:29.243127: I tensorflow/core/platform/cpu_feature_guard.cc:142] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2
2020-01-13 16:01:29.256487: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1555] Found device 0 with properties:
pciBusID: 0000:02:00.0 name: Quadro M5000 computeCapability: 5.2
coreClock: 1.038GHz coreCount: 16 deviceMemorySize: 7.50GiB deviceMemoryBandwidth: 196.99GiB/s
2020-01-13 16:01:29.269781: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cudart64_101.dll
2020-01-13 16:01:29.275906: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cublas64_10.dll
2020-01-13 16:01:29.285202: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cufft64_10.dll
2020-01-13 16:01:29.291266: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library curand64_10.dll
2020-01-13 16:01:29.299261: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cusolver64_10.dll
2020-01-13 16:01:29.304025: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cusparse64_10.dll
2020-01-13 16:01:29.311939: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cudnn64_7.dll
2020-01-13 16:01:29.316594: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1697] Adding visible gpu devices: 0
2020-01-13 16:01:30.029865: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1096] Device interconnect StreamExecutor with strength 1 edge matrix:
2020-01-13 16:01:30.036285: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1102]      0
2020-01-13 16:01:30.040429: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1115] 0:   N
2020-01-13 16:01:30.045089: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1241] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 6016 MB memory) -> physical GPU (device: 0, name: Quadro M5000, pci bus id: 0000:02:00.0, compute capability: 5.2)
WARNING:tensorflow:AutoGraph could not transform <function test at 0x000001977DE52438> and will run it as-is.
Please report this to the TensorFlow team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output.
Cause: Bad argument number for Name: 3, expecting 4
```

**Comment by bersbersbers** [comment]
I believe this was related to having `gast==0.3.2` installed.

**Comment by bersbersbers** [comment]
Having `gast==0.3.2` installed (by accident) can cause a host of AutoGraph-related errors. Here's another example - maybe this is helpful to someone:

> OperatorNotAllowedInGraphError: using a `tf.Tensor` as a Python `bool` is not allowed: AutoGraph is disabled in this function. Try decorating it directly with @tf.function.
```
import tensorflow as tf

def test(data):
    if tf.random.uniform(()) < 0.5:
        pass
    return data

tf.data.Dataset.from_tensor_slices([[[0]]]).map(test)
```
This is also fixed by `pip install gast==0.2.2`  for TF2.1.
